linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,258 @@
+# threadpool_linux
+
+## Overview
+
+`threadpool_linux` is the Linux-specific implementation of the `threadpool` module.
+
+## Design
+
+`threadpool_linux` provides the Linux implementation of `threadpool`. The `threadpool` object contains an array of threads all initialized in `threadpool_open_async`  and an array of tasks all initialized in `threadpool_create`, all incoming tasks are being added in `threadpool_schedule_work` and scheduled in `threadpool_work_func`. 
+
+`threadpool` maintains the following states:
+1. `state`: a 32-bit variable represents a state enum value, which contains: 
+    - `THREADPOOL_STATE_NOT_OPEN` : entered after a call to `threadpool_create` or a call to `internal_close`.
+    - `THREADPOOL_STATE_OPENING` : entered in the beginning of a call to `threadpool_open_async`.
+    - `THREADPOOL_STATE_OPEN` : entered after the call to `threadpool_open_async`.
+    - `THREADPOOL_STATE_CLOSING` : entered in the beginning of a call to `internal_close`.
+2. `max_thread_count`, `min_thread_count`, `used_thread_count` : a 32-bit counters for the `threadpool`.
+3. `semaphore` : a semaphore to ensure the number of waiting tasks changed automically when a task is inserted or consumed in the array.
+4. `task_array` : a circular array of waiting tasks with default size 2048, initialized in `threadpool_create`. Each array item contains a `THREADPOOL_TASK` object, which obtains the `task_state` and task function information.
+    - `task_state` can be following:
+       - `TASK_NOT_USED` : entered after a call to `threadpool_create`, new array items after a call to `reallocate_threadpool_array`, and a call to `threadpool_work_func`.
+       - `TASK_INITIALIZING` : entered in the progress of a call to `threadpool_schedule_work`.
+       - `TASK_WAITING` : entered after a call to `threadpool_schedule_work` with a zero return value.
+       - `TASK_WORKING` : entered in the progress of a call to `threadpool_work_func` by calling `threadpool_open_async`.
```

Comment:
Let's add what the state means, rather than just mentioning the transitions.
Like: 
"TASK_NOT_USED - Entry is not used. Entered after a call to `threadpool_create`, new array items after a call to `reallocate_threadpool_array`, and a call to `threadpool_work_func`."

Reply:
added